src/main/java/com/example/Library/controller/BooksController.java

```diff
@@ -13,36 +15,38 @@
 import java.util.Optional;
 
 @RestController
-@RequestMapping("api/v1/Admins")
-public class AdminRestController {
+@RequestMapping("/api/v1/Books")
+public class BooksController {
+
     private final BookService bookService;
 
     @Autowired
-    public AdminRestController(BookService bookService) {
+    public BooksController(BookService bookService) {
         this.bookService = bookService;
     }
 
-    @GetMapping("/getAllBook")
-    @PreAuthorize("hasAuthority('user:admin')")
+    @PreAuthorize("hasAuthority('user:guest')")
+    @GetMapping()
     public ArrayList<Book> getAllBook() {
         return (ArrayList<Book>) bookService.getAllBooks();
     }
 
-    @PreAuthorize("hasAuthority('user:admin')")
-    @GetMapping("/getBookByTitle/{title}")
-    public Book getBookByTitle(@PathVariable String title) {
-        return bookService.getBookByTitle(title)
-                .orElseThrow(() -> new BookNotFoundByTitleException(title));
-    }
-    @PreAuthorize("hasAuthority('user:admin')")
-    @GetMapping("/getBookById/{id}")
+    @PreAuthorize("hasAuthority('user:guest')")
+    @GetMapping("/{id}")
     public Book getBookById(@PathVariable long id) {
         return bookService.getBookById(id)
                 .orElseThrow(() -> new BookNotFoundByIdException(id));
     }
 
+    @PreAuthorize("hasAuthority('user:guest')")
+    @GetMapping("getBookByTitle/{title}")
```

Comment:
it would be better to replace "getBookByTitle/{title}" with 'title' query param + base resource name, like  described in Filtering section of https://documenter.getpostman.com/view/1372588/S1LsZqc9#17aff2e0-af6e-4876-8bd6-adf429d2c81b